### PR TITLE
Make cross-project step preflights work

### PR DESCRIPTION
**Pull request by davidmreed**
# Critical Changes

# Changes

# Issues Closed

- We fixed an issue preventing step-level preflight checks from working correctly in MetaDeploy when run as a cross-project flow.
- The `metadeploy_publish` task now displays a clear error message if you supply the wrong API URL.

### CODE REVIEW

File: cumulusci/core/flowrunner.py [hunk 1]
```diff
@@ -790,7 +802,7 @@ def __call__(self, **options):
         if cache_key in self.cache.results:
             return self.cache.results[cache_key].return_values
 
-        task_config = self.cache.flow.project_config.tasks[self.task_name]
+        task_config = self.cache.project_config.tasks[self.task_name]
```
Comment (davisagli) [inline cumulusci/core/flowrunner.py:805]: @davidmreed we should also pass the step-specific project config below on line 812

**Review by davisagli — CHANGES_REQUESTED** [review]
Good analysis and fix. I think I see one more place where we need to pass the step-specific project_config

File: cumulusci/core/flowrunner.py (hunk 1 above)
Comment (davidmreed) [reply cumulusci/core/flowrunner.py:805]: 👍 Good catch.

**Review by davisagli — APPROVED** [review]

**Comment by davisagli** [comment]
Hmm, what's up with the failed test in py3.8?

**Comment by davidmreed** [comment]
It passed on a retry; looks like maybe a flappy Snowfakery user channels test. @prescod Do you recognize it?

**Comment by davidmreed** [comment]
I'll go ahead and merge since I don't see any plausible link to this work.

**Pull request merged**